pkg/loaders/azure/models/job.go

```diff
@@ -0,0 +1,168 @@
+package models
+
+import (
+	"github.com/argonsecurity/pipeline-parser/pkg/consts"
+	loadersUtils "github.com/argonsecurity/pipeline-parser/pkg/loaders/utils"
+	"github.com/argonsecurity/pipeline-parser/pkg/models"
+	"gopkg.in/yaml.v3"
+)
+
+type JobType string
+
+const (
+	CIJobType         JobType = "job"
+	DeploymentJobType JobType = "deployment"
+	TemplateJobType   JobType = "template"
+)
+
+type Workspace struct {
+	Clean string `yaml:"clean,omitempty"`
+}
+
+type Uses struct {
+	Repositories []string `yaml:"repositories,omitempty"`
+	Pools        []string `yaml:"pools,omitempty"`
+}
+
+type BaseJob struct {
+	DisplayName            string            `yaml:"displayName,omitempty"`
+	DependsOn              []string          `yaml:"dependsOn,omitempty"`
+	Condition              string            `yaml:"condition,omitempty"`
+	ContinueOnError        bool              `yaml:"continueOnError,omitempty"`
+	TimeoutInMinutes       int               `yaml:"timeoutInMinutes,omitempty"`
+	CancelTimeoutInMinutes int               `yaml:"cancelTimeoutInMinutes,omitempty"`
+	Variables              *Variables        `yaml:"variables,omitempty"`
+	Pool                   *Pool             `yaml:"pool,omitempty"`
+	Container              *JobContainer     `yaml:"container,omitempty"`
+	Services               map[string]string `yaml:"services,omitempty"`
+	Workspace              *Workspace        `yaml:"workspace,omitempty"`
+	Uses                   *Uses             `yaml:"uses,omitempty"`
+	Steps                  *Steps            `yaml:"steps,omitempty"`
+	TemplateContext        map[string]any    `yaml:"templateContext,omitempty"`
+}
+
+type CIJob struct {
+	Job           string       `yaml:"job,omitempty"`
+	Strategy      *JobStrategy `yaml:"strategy,omitempty"`
+	BaseJob       `yaml:",inline"`
+	FileReference *models.FileReference
+}
+
+type DeploymentEnvironment struct {
+	Name          string `yaml:"name,omitempty"`
+	ResourceName  string `yaml:"resourceName,omitempty"`
+	ResourceId    string `yaml:"resourceId,omitempty"`
+	ResourceType  string `yaml:"resourceType,omitempty"`
+	Tags          string `yaml:"tags,omitempty"`
+	FileReference *models.FileReference
+}
+
+func (de *DeploymentEnvironment) UnmarshalYAML(node *yaml.Node) error {
+	de.FileReference = loadersUtils.GetFileReference(node)
+	if node.Tag == consts.StringTag {
+		de.Name = node.Value
+		return nil
+	}
+
+	return node.Decode(&de)
+}
+
+type DeploymentJob struct {
+	Deployment    string                 `yaml:"deployment,omitempty"`
+	Strategy      *DeploymentStrategy    `yaml:"strategy,omitempty"`
+	Environment   *DeploymentEnvironment `yaml:"environment,omitempty"`
+	BaseJob       `yaml:",inline"`
+	FileReference *models.FileReference
+}
+
+type TemplateJob struct {
+	Template      `yaml:",inline"`
+	FileReference *models.FileReference
+}
+
+type Jobs struct {
```

Comment:
It's good practice to use an array of pointers, rather than a pointer of an array